### PR TITLE
contrib/openvpn: new package (2.6.8)

### PR DESCRIPTION
**Pull request by vakhrushevdv**

### CODE REVIEW

File: contrib/openvpn/template.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    "--enable-x509-alt-username",
+]
+make_dir = "."
+hostmakedepends = [
+    "automake",
+    "pkgconf",
+    "libtool",
+    "python",
+    "openssl",
+    "cmocka",
```
Comment (nekopsykose) [inline contrib/openvpn/template.py:17]:
```suggestion
```

cmocka here doesn't do anything, the openssl probably neither unless it calls `openssl(1)` in the build

File: contrib/openvpn/template.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+pkgrel = 0
+build_style = "configure"
+configure_args = [
+    "--with-openssl-engine",
+    "--with-crypto-library=openssl",
+    "--enable-x509-alt-username",
+]
+make_dir = "."
+hostmakedepends = [
+    "automake",
```
Comment (nekopsykose) [inline contrib/openvpn/template.py:12]: sort these *depends lists

File: contrib/openvpn/template.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
+pkgname = "openvpn"
+pkgver = "2.6.8"
+pkgrel = 0
+build_style = "configure"
```
Comment (nekopsykose) [inline contrib/openvpn/template.py:4]:
```suggestion
build_style = "gnu_configure"
```

File: contrib/openvpn/template.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+configure_args = [
+    "--with-openssl-engine",
+    "--with-crypto-library=openssl",
+    "--enable-x509-alt-username",
+]
+make_dir = "."
```
Comment (nekopsykose) [inline contrib/openvpn/template.py:10]: haven't checked but most gnu_configure projects don't need this

File: contrib/openvpn/template.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+    "libcap-ng-devel",
+    "cmocka-devel",
+]
+pkgdesc = "Secure tunneling daemon"
+maintainer = "Dmitriy Vakhrushev <dvakhrushev@netgate.com>"
+license = "GPL-2.0-only"
+url = "https://github.com/OpenVPN/openvpn"
+source = f"https://github.com/OpenVPN/openvpn/releases/download/v{pkgver}/{pkgname}-{pkgver}.tar.gz"
+sha256 = "5ede1565c8a6d880100f7f235317a7ee9eea83d5052db5547f13a9e76af7805d"
+options = ["!check"]
```
Comment (nekopsykose) [inline contrib/openvpn/template.py:35]: need a comment for why

File: contrib/openvpn/template.py [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+    "lzo-devel",
+    "lz4-devel",
+    "libnl-devel",
+    "libcap-ng-devel",
+    "cmocka-devel",
+]
+pkgdesc = "Secure tunneling daemon"
+maintainer = "Dmitriy Vakhrushev <dvakhrushev@netgate.com>"
+license = "GPL-2.0-only"
+url = "https://github.com/OpenVPN/openvpn"
```
Comment (nekopsykose) [inline contrib/openvpn/template.py:32]:
```suggestion
url = "https://openvpn.net"
```

File: contrib/openvpn/template.py [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+    "pcre2-devel",
+    "lzo-devel",
+    "lz4-devel",
+    "libnl-devel",
+    "libcap-ng-devel",
+    "cmocka-devel",
+]
+pkgdesc = "Secure tunneling daemon"
+maintainer = "Dmitriy Vakhrushev <dvakhrushev@netgate.com>"
+license = "GPL-2.0-only"
```
Comment (nekopsykose) [inline contrib/openvpn/template.py:31]:
```suggestion
license = "GPL-2.0-only WITH openvpn-openssl-exception"
```

File: contrib/openvpn/template.py [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+hostmakedepends = [
+    "automake",
+    "pkgconf",
+    "libtool",
+    "python",
+    "openssl",
+    "cmocka",
+]
+makedepends = [
+    "linux-pam-devel",
```
Comment (nekopsykose) [inline contrib/openvpn/template.py:20]: this list also needs linux-headers. also i think you need `depends = ` on iproute2 (it needs `ip` right?)

**Comment by nekopsykose** [comment]
you also need a service file for this, which is probably not that easy to write

**Comment by triallax** [comment]
we merged in https://github.com/chimera-linux/cports/pull/2524 which adds openvpn recently, but thanks for taking the time to submit this anyway!

**Pull request closed without merging**